packages/discovery-react-components/src/components/DiscoverySearch/DiscoverySearch.tsx

```diff
@@ -480,14 +484,46 @@ const DiscoverySearch: FC<DiscoverySearchProps> = ({
   );
 
   const handleFetchDocuments = useCallback(
-    (filterString: string, searchResponse: DiscoveryV2.QueryResponse | null): void => {
-      fetchDocuments(filterString, filteredResponse => {
+    (
+      filterString: string,
+      collections?: string[],
+      searchResponse?: DiscoveryV2.QueryResponse
+    ): void => {
+      fetchDocuments(filterString, collections || [], filteredResponse => {
         if (searchResponse && searchResponse.results && filteredResponse.results) {
+          const results = [...searchResponse.results];
+          const filteredResults = [...filteredResponse.results];
+
+          // replace existing doc data in `searchResponse` with new filtered response data
+          while (filteredResults.length) {
+            const filteredResult = filteredResults.pop();
+            if (!filteredResult) {
+              continue;
+            }
+
+            const idx = results.findIndex(
+              res =>
+                res.document_id === filteredResult?.document_id &&
+                res.result_metadata.collection_id === filteredResult.result_metadata.collection_id
+            );
+
+            if (idx !== -1) {
+              // merge results, since filteredResult may not contain all fields from a previous
+              // query (like passages)
+              const mergedResult = {
+                ...results[idx],
+                ...filteredResult
+              };
+              results.splice(idx, 1, mergedResult);
+            } else {
+              results.push(filteredResult);
+            }
+          }
+
           setSearchResponse({
```

Comment:
For some reason that I don't know, `fetchDocument()` updates the search response store.

There are several "stores" handled by this code, two of which are:
- the `searchResponseStore`, which is updated by `performSearch()`
- the `fetchDocumentsResponseStore`, which is updated by `fetchDocument()`

But `fetchDocument()` also updates `searchResponseStore`. Why?

Best I can figure is something like the scenario that this PR enables: perform a search but only request the minimal data required (which ends up in the `searchResponseStore`). But then we want to display then entire document, so we call `fetchDocument()` which returns the full doc data. So may as well add this new data to the existing one in `searchResponseStore`.

But I think that should be an option for `fetchDocument()`, not the default. That is, by default, calling the method will only update the `fetchDocumentsResponseStore`. If you also want to update the `searchResponseStore`, the pass a boolean option. Thoughts?